### PR TITLE
Created documentation for kustomization deployment

### PR DESCRIPTION
**Pull request by l-maciej**
Initial commit for kustomization-based deployment.

### CODE REVIEW

File: docs/install.md [hunk 1]
```diff
@@ -185,8 +185,6 @@ ntfy is packaged in nixpkgs as `ntfy-sh`. It can be installed by adding the pack
 nix-env -iA ntfy-sh
 ```
 
-NixOS also supports [declarative setup of the ntfy server](https://search.nixos.org/options?channel=unstable&show=services.ntfy-sh.enable&from=0&size=50&sort=relevance&type=packages&query=ntfy). 
-
```
Comment (binwiederhier) [inline docs/install.md:189]: Your branch seems to be out of date. Pull the latest changes via `git pull upstream main` ("upstream" is the remote https://github.com/binwiederhier/ntfy, Google `git remote add`).

File: docs/install.md [hunk 2]
```diff
@@ -300,12 +298,12 @@ This image can be pushed to a container registry and shipped independently. All
...
-are a few options to mix and match, including a deployment without a cache file, a stateful set with a persistant cache, and a standalone
+are a few options to mix and match, including a deployment without a cache file, a stateful set with a persistent cache, and a standalone
 unmanaged pod.
 
-
-=== "deployment"
-    ```yaml
+Deployment
+===
+  ```yaml
```
Comment (binwiederhier) [inline docs/install.md:306]: While not correct Markdown, this syntax is is actually correct for `mkdocs`. It generates these abs that you see in the docs everywhere (example: https://ntfy.sh/docs/publish/).

So please revert this.

File: docs/install.md [hunk 3]
```diff
@@ -350,10 +348,12 @@ unmanaged pod.
...
         targetPort: 80
-    ```
+  ```
+
 
-=== "stateful set"
-    ```yaml
+Stateful set
+=== 
+  ```yaml
```
Comment (binwiederhier) [inline docs/install.md:356]: Same

File: docs/install.md [hunk 4]
```diff
@@ -391,10 +391,12 @@ unmanaged pod.
           resources:
             requests:
               storage: 1Gi
-    ```
+  ```
+
+Pod
+===
```
Comment (binwiederhier) [inline docs/install.md:397]: Same

File: docs/install.md [hunk 5]
```diff
@@ -420,12 +422,15 @@ unmanaged pod.
...
+```
 
-Configuration is relatively straightforward. As an exmaple, a minimal configuration is provided.
+Configuration is relatively straightforward. As an example, a minimal configuration is provided.
 
-=== "resource definition"
-    ```yaml
+
+Resource definition
+===
```
Comment (binwiederhier) [inline docs/install.md:431]: Same

File: docs/install.md [hunk 6]
```diff
@@ -434,9 +439,179 @@ Configuration is relatively straightforward. As an exmaple, a minimal configurat
...
+
+from-file
+=== 
+
+  ```bash
     kubectl create configmap ntfy --from-file=server.yml 
-    ```
+  ```
+
+Kustomize-based deployment with persistent storage and Traefik ingress
```
Comment (binwiederhier) [inline docs/install.md:452]: 1. The section should just be called "Kustomize". The sidebar titles should be short
2. The whole section is a little long and has many rough edges. Please make sure it looks similar to the other install instructions when you generate the docs (mkdocs serve). I can probably help with this, but right now the PR is too rough to even do that.

**Review by binwiederhier — CHANGES_REQUESTED** [review]

**Comment by binwiederhier** [comment]
:+1: Thank you for your PR. I left some comments, which you'd need to address before I can merge it. Usually I just merge it and then do the format fixes later, but the PR is too extensive to do that myself.

File: docs/install.md (hunk 2 above)
Comment (l-maciej) [reply docs/install.md]: Reverted to mkdocs syntax

File: docs/install.md (hunk 3 above)
Comment (l-maciej) [reply docs/install.md]: Reverted to mkdocs syntax

File: docs/install.md (hunk 4 above)
Comment (l-maciej) [reply docs/install.md]: Reverted to mkdocs syntax

File: docs/install.md (hunk 5 above)
Comment (l-maciej) [reply docs/install.md]: Reverted to mkdocs syntax

File: docs/install.md (hunk 6 above)
Comment (l-maciej) [reply docs/install.md]: Section is cleaned up, installation steps and some references are added.

**Comment by binwiederhier** [comment]
Your branch is still not up-to-date, and you've change permissions on various files (see "Files" tab in this PR).

Do this to update your branch:
```
git remote add upstream git@github.com:binwiederhier/ntfy.git
git fetch upstream
git pull upstream main
git push origin main
```

You'll still need to revert the file permissions.

**Comment by binwiederhier** [comment]
Ping.

**Comment by Ali2731** [comment]
Your branch is still not up-to-date, and you've change permissions on various files (see "Files" tab in this PR).

Do this to update your branch:

git remote add upstream git@github.com:binwiederhier/ntfy.git
git fetch upstream
git pull upstream main
git push origin main
You'll still need to revert the file permissions.

**Comment by l-maciej** [comment]
Hi I've reverted permissions and pulled in fresh changes to the branch main branch.

**Comment by binwiederhier** [comment]
Almost there. If you look at the diff https://github.com/binwiederhier/ntfy/pull/463/files, you'll see that the is a string that your PR removes that is entirely unrelated. That should be fixed by pulling from upstream as described above.

If you can't figure out how, I can do it :)

File: docs/install.md (hunk 1 above)
Comment (l-maciej) [reply docs/install.md]: Line restored

**Comment by binwiederhier** [comment]
Looks good. Thanks.

**Pull request merged**